packages/malloy/src/dialect/snowflake/snowflake.ts

```diff
@@ -266,7 +266,16 @@ export class SnowflakeDialect extends Dialect {
       parentType === 'array[scalar]' ||
       parentType === 'array[record]'
     ) {
-      const arrayRef = `"${parentAlias}".value:${sqlName}`;
+      // Case 1: this is an array of scalar. We can reference the field
```

Comment:
I'm not 100% sure I know what is going because I don't have the un-nested state of the array exactly in my head at this moment, but something like this would be preferable, if this is the correct solution.

```TypeScript
      const arrayRef = `"${parentAlias}".value` + (parentType === 'array[record]' ? `:${sqlName }`: '');
```